### PR TITLE
summary: Fix hparams name display for Callables

### LINKED ISSUE #579: hparams: Bug on displaying callable hyperparameters name
**Describe the bug**
A hyperparameter may not always be an `int`, `str`, `bool`, `float` or `None`. In my case, I have callables. For example, doing a hyperparameter optimization on final activations or even loss functions. The experiment runs fine (thanks to 9d267066) but on my `loss` column in `hparams` in Tensorboard, all fields under that are blank.

**Expected behavior**
If the param is a callable, take its name for proper display.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by ThibzR**
fix #579 

This PR fixes the issue by taking the `__name__` field of the callable if it exists, falling back on `str()` when it does not. Tested it on my projects, looks nice!

After fix:
image

The "before" screenshot is available in #579

### CODE REVIEW

File: tensorboardX/proto/api.proto [hunk 1]
```diff
@@ -104,6 +104,7 @@ enum DataType {
   DATA_TYPE_STRING = 1;
   DATA_TYPE_BOOL = 2;
   DATA_TYPE_FLOAT64 = 3;
+  DATA_TYPE_CALLABLE = 4;
```
Comment (lanpa) [inline tensorboardX/proto/api.proto:107]: I think we should stay on DATA_TYPE_STRING, this additional DataType might conflict with the official TensorBoard if they adds another type in the future.

**Comment by ThibzR** [comment]
@lanpa Makes sense! I have removed changes from `api.proto` and fell back on `DATA_TYPE_STRING` for callables

**Comment by lanpa** [comment]
Thanks!
